### ISSUE TITLE
Ability to use the Boss CLI interface

### ISSUE THREAD

**Issue by kabirbaidhya** [kabirbaidhya/boss#20]
Ability to use `boss` cli tool instead of just using the `fab` command

**Comment by srishanbhattarai** [comment]
I can help here. How do you want to do this?

We can start with rudimentary support by piping every command through to `fab`. So,
`boss [variadic]` --> `fab [variadic]` would be a simple starting point. 

We can also have POSIX style custom CLI commands.
`boss deploy --stage=dev` etc.

**Comment by kabirbaidhya** [comment]
@srishanbhattarai Firstly, thanks for offering to help.

I was thinking of doing the exactly same thing before i.e piping everything from `boss` to `fab` cli. For instance being able to do `boss prod deploy` instead of `fab prod deploy`. But later I realized it's not simply possible without a weird hack and I did rollback. 

Also, looking at the [issues](https://github.com/fabric/fabric/issues) and [PRs](https://github.com/fabric/fabric/pulls) pending on fabric seems like it isn't that well maintained either. 

And the [lack of python3 support](https://travis-ci.org/kabirbaidhya/boss/builds/291165785), is creating another bottleneck on our side due to fabric. Which is why I'm thinking of making it independent of fabric at least where ever possible right now, and later get rid of fabric by just relying on `paramiko` internally.

So, I'm thinking of having our own set of commands for boss something like these:
```
boss init [--interactive]
boss deploy <stage>
boss rollback <stage>
boss run <script or task>...
boss ls or boss list
```
Optionally, we can add POSIX style options too.
```
boss deploy --stage=prod
boss run <task or script> --stage=prod
```
And so on. 

Anyway, I'll create another issue with a checklist of what is/needs to be done for the CLI and you can choose which one you'd like to help in.

**Comment by kabirbaidhya** [comment]
@srishanbhattarai I've created a new issue https://github.com/kabirbaidhya/boss/issues/74 for cli. Let me know if there's anything in the list you'd like to contribute to, or have any more suggestions.
